src/test/compile-fail/hygiene/fields.rs

```diff
@@ -22,11 +22,11 @@ mod foo {
             x: i32,
         }
 
-        let s = S { x: 0 };
-        let _ = s.x;
+        let s = S { x: 0 }; //~ ERROR type `foo::S` is private
```

Comment:
Macro `m` leaks a value of private type `S`, e.g.  the expanded code contains value of private type `S`.

This specific case shouldn't necessarily be an error, because `S` is used "internally" in the macro, and the macro doesn't "return" it to the code outside of it. Discerning between macros using something internally or not is a separate problem that may be solved in the future (or [may be not](https://github.com/rust-lang/rust/pull/42125#issuecomment-304475760)), so this PR implements the conservative solution - everything used in a macro is considered leaked (this is mentioned in the PR description).